### PR TITLE
Added Gas Canisters [Unfinished]

### PR DESCRIPTION
**Pull request by clement-or**
I added some features for gas canisters with a placeholder UI.

For now : 
* Canisters can contain gases
* Canisters can release gases
* Canister can be attached to ports, tho they are not implemented yet
* Canisters can be re-labeled

I'm making this PR because I won't be able to pursue work on this for at least 2 months, so someone could finish the missing features, because canisters are an important part to the game.

### CODE REVIEW

File: Resources/Credits/GitHub.txt [hunk 1]
```diff
@@ -1 +1 @@
...
...TheSystem, CC-4477, Centronias, chairbender, Topy, clusterfack, Clyybber, collinlunn, ComicIronic, creadth, Cyberboss, d34d10cc, DamianX, Daracke, DeanUA, deathrat, Decappi, Dezandor, DrSmugleaf, DTanxxx, dylanstrategie, exp111, F77F, FirinMaLazors, fizzle7, Fouin, gbasood, gituhabu, GlassEclipse, h3half, halworsen, harikattar, Hugal31, ike709, Injazz, InquisitivePenguin, j-giebel, Jackw2As, janekvap, JiimBob, johnku1, juliangiebel, kalanosh, Kmc2000, lajolico, lzimann, Macoron, MemeProof, metalgearsloth, micheel665, Miniwoffer, Moneyl, N3X15, NickPowers43, nuke-makes-games, Nukeuler123, nullarmo, Pangogie, partyaddict, patrikturi, PaulRitter, peptron1, Pireax, PJB3005, ProfanedBane, PrPleGoo, psykzz, Qustinnus, RednoWCirabrab, Rember, RemieRichards, remove32, renodubois, rneuser, Rockdtben, Rohesie, rok-povsic, RomanNovo, SamV522, ScumbagDog, sentient-ai, Serkket, ShadowCommander, SignalWalker, Silvertorch5, SoulSloth, Soundwavesghost, SpaceManiac, spoogemonster, StrawberryMoses, SweptWasTaken, Szunti, tentekal, thatrandomcanadianguy, TheCze, theOperand, tkdrg, Tomeno, Tyler-IN, unusualcrow, UristMcDorf, Visne, volundr-, wixoaGit, YotaXP, zamp, ZelteHonor, ZNixian, Zth--, Zumorica
```
Comment (SweptWasTaken) [inline Resources/Credits/GitHub.txt:1]: 👀

Comment (clement-or) [reply Resources/Credits/GitHub.txt]: Whoops

**Comment by 20kdc** [comment]
I'm taking on maintenance/work on this PR under the new ID #2628. If this is a problem, please alert me and I'll close it.

**Comment by Zumorica** [comment]
Closing in favor of #2628 for now

**Pull request closed without merging**